dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/PagingCriteriaProcessor.java

```diff
@@ -43,19 +43,17 @@
 @RequiredArgsConstructor
 public class PagingCriteriaProcessor implements Processor<AnalyticsPagingCriteria>
 {
-
     private final SystemSettingManager systemSettingManager;
 
     // TODO: DHIS2-13384 we would really like to have all
     // criteria/request/params to be
     // immutable, but PagingCriteria is not
     // returning it for now, should be converted to use builders
     @Override
-    public AnalyticsPagingCriteria process( final AnalyticsPagingCriteria pagingCriteria )
+    public AnalyticsPagingCriteria process( AnalyticsPagingCriteria pagingCriteria )
```

Comment:
I realized that a few `final keywords were removed. I usually use it everywhere to make it clear they are immutable.
Personally, I found it easier to read and a good way to avoid side effects.
If you guys prefer to not use `final`, please let me know. I will stop using it and will start removing them instead =D

Reply:
Cool. It is fine per se, sorry for removing. In this case, it does not really make the argument immutable, as the inner state (properties) of `AnalyticsPagingCriteria` could still be changed, so it would give sort of a false sense of immutability.